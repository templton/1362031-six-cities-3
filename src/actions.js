// Удалить полностью

const ActionType = {
  SET_CURRENT_CITY: `SET_CURRENT_CITY`,
  SORT_PLACES_IN_CITY: `SORT_PLACES_IN_CITY`,
  SET_MOUSE_OVER_PLACE_CARD: `SET_MOUSE_OVER_PLACE_CARD`,
  UNSET_MOUSE_OVER_PLACE_CARD: `UNSET_MOUSE_OVER_PLACE_CARD`,
};

const sortCityFilterType = {
  POPULAR: `Popular`,
  TOP_RATED_FIRST: `Top rated first`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
};

const setCurrentCity = (currentCity) => ({type: ActionType.SET_CURRENT_CITY, payload: currentCity});

const sortPlacesInCity = (filterType) => ({type: ActionType.SORT_PLACES_IN_CITY, payload: filterType});

const setMouseOverPlaceCard = (card) => ({type: ActionType.SET_MOUSE_OVER_PLACE_CARD, payload: card});
const unsetMouseOverPlaceCard = () => ({type: ActionType.UNSET_MOUSE_OVER_PLACE_CARD});

export {setCurrentCity, sortPlacesInCity, setMouseOverPlaceCard, unsetMouseOverPlaceCard, ActionType, sortCityFilterType};
