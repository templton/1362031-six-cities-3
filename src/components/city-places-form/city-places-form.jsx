import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import CityPlacesFormFilters from "../city-places-form-filters/city-places-form-filters";
import {selectCurrentPlacesFilter} from "../../reducer/places-filter/selectors";

const CityPlacesForm = ({currentPlaceFilterType, isToggleElementActive, handleToggle}) => {

  return (
    <form className="places__sorting" action="#" method="get">
      <span className="places__sorting-caption">Sort by</span>
      <span className="places__sorting-type" onClick={handleToggle}>
      &nbsp; {currentPlaceFilterType}
        <svg className="places__sorting-arrow" width="7" height="4">
          <use xlinkHref="#icon-arrow-select"></use>
        </svg>
      </span>
      <CityPlacesFormFilters isFilterVisible={isToggleElementActive} onFilterClick={handleToggle}/>
    </form>
  );
};

CityPlacesForm.propTypes = {
  currentPlaceFilterType: PropTypes.string.isRequired,
  isToggleElementActive: PropTypes.bool.isRequired,
  handleToggle: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  currentPlaceFilterType: selectCurrentPlacesFilter(state)
});

export {CityPlacesForm};
export default connect(mapStateToProps)(CityPlacesForm);
