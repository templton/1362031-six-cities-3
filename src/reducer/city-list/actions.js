const ActionType = {
  LOAD_CITY_LIST: `LOAD_CITY_LIST`
};

const ActionCreator = {
  loadCityList(payload) {
    return ({type: ActionType.LOAD_CITY_LIST, payload});
  }
};

export {ActionCreator, ActionType};
