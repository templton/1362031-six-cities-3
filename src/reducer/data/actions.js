const ActionType = {
  LOAD_DATA: `LOAD_DATA`
};

const ActionCreator = {
  loadData(payload) {
    return {type: ActionType.LOAD_DATA, payload}
  }
};

export {ActionType, ActionCreator};
