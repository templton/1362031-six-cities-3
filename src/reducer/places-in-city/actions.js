const ActionType = {
  LOAD_PLACES_IN_CITIES: `LOAD_PLACES_IN_CITIES`
};

const ActionCreator = {
  loadPlacesInCity(payload) {
    return {type: ActionType.LOAD_PLACES_IN_CITIES, payload};
  }
};

export {ActionType, ActionCreator};
