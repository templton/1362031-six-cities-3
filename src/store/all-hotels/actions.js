const ActionType = {
  LOAD_ALL_HOTELS: `LOAD_ALL_HOTELS`,
  SET_LOADING: `SET_LOADING`,
  SET_NEARBY_HOTELS: `SET_NEARBY_HOTELS`,
};

const ActionCreator = {
  loadAllHotels(hotels) {
    return {type: ActionType.LOAD_ALL_HOTELS, payload: hotels};
  },
  setLoading(status) {
    return {type: ActionType.SET_LOADING, payload: status};
  },
  setNearbyHotels(nearbyPlaces) {
    return {type: ActionType.SET_NEARBY_HOTELS, payload: nearbyPlaces};
  }
};

export {ActionType, ActionCreator};
