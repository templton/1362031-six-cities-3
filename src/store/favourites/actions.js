const ActionType = {
  LOAD_FAVOURITES: `LOAD_FAVOURITES`
};

const ActionCreator = {
  loadFavourites: (payload) => ({type: ActionType.LOAD_FAVOURITES, payload})
};

export {ActionType, ActionCreator};
