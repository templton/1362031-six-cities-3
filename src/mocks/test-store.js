import {sortCityFilterType} from "../actions";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const offers = [
  {
    id: 1,
    title: `Комната №1`,
    isPremium: false,
    image: `https://upload.wikimedia.org/wikipedia/commons/3/34/Nanteos_%28geograph_3153536%29.jpg`,
    costPerNignt: 80,
    type: `room`,
    rating: 4.7,
    cord: [52.3909553943508, 4.90309666406198]
  },
  {
    id: 2,
    title: `Комната №115`,
    isPremium: false,
    image: `https://upload.wikimedia.org/wikipedia/commons/3/34/Nanteos_%28geograph_3153536%29.jpg`,
    costPerNignt: 60,
    type: `apartment`,
    rating: 2,
    cord: [52.3909553943508, 4.90309666406198]
  },
];


const city = {
  id: 17,
  name: `Амстердам`,
  cord: [52.38333, 4.9]
};

const cities = [
  {
    id: 17,
    name: `Амстердам`,
    cord: [52.38333, 4.9]
  },
  {
    id: 15,
    name: `Барнаул`,
    cord: [53.346785, 83.776856]
  },
  {
    id: 16,
    name: `Новосибирск`,
    cord: []
  }
];

const store = mockStore({
  citiesList: cities,
  placesInCity: offers,
  city,
  currentPlaceFilterType: sortCityFilterType.POPULAR
});

export {store};
