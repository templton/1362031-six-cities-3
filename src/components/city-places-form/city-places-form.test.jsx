import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {sortCityFilterType} from "../../actions";
import {CityPlacesForm} from "./city-places-form";
import {store} from "../../mocks/test-store";


describe(`CiltyPlacesForm render`, () => {
  it(`Simple render`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <CityPlacesForm handleToggle={()=>{}} isToggleElementActive={false} currentPlaceFilterType={sortCityFilterType.POPULAR}/>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
