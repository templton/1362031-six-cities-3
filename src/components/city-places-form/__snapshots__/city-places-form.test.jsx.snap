// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CiltyPlacesForm render Simple render 1`] = `
<form
  action="#"
  className="places__sorting"
  method="get"
>
  <span
    className="places__sorting-caption"
  >
    Sort by
  </span>
  <span
    className="places__sorting-type"
    onClick={[Function]}
  >
    Â  
    Popular
    <svg
      className="places__sorting-arrow"
      height="4"
      width="7"
    >
      <use
        xlinkHref="#icon-arrow-select"
      />
    </svg>
  </span>
</form>
`;
